truffle-config.js

```diff
@@ -1,18 +1,126 @@
-/*
- * NB: since truffle-hdwallet-provider 0.0.5 you must wrap HDWallet providers in a 
- * function when declaring them. Failure to do so will cause commands to hang. ex:
- * ```
- * mainnet: {
- *     provider: function() { 
- *       return new HDWalletProvider(mnemonic, 'https://mainnet.infura.io/<infura-key>') 
- *     },
- *     network_id: '1',
- *     gas: 4500000,
- *     gasPrice: 10000000000,
- *   },
+/**
+ * Use this file to configure your truffle project. It's seeded with some
+ * common settings for different networks and features like migrations,
+ * compilation and testing. Uncomment the one's you need or modify
+ * them to suit your project as necessary.
+ *
+ * This file is named `truffle-config` for Windows users because `truffle.js`
+ * conflicts with the `truffle` command on that platform. (You can also
+ * use this file on Linux and Mac.)
+ *
+ * More information about configuration can be found throughout the
+ * truffle docs and especially at:
+ *
+ * truffleframework.com/docs/advanced/configuration
+ *
  */
 
+
+/**
+ * If you'd like to deploy via Infura, you'll need an HD wallet provider.
+ *
+ * > Using Truffle V5 or later? Make sure you install the V5 version.
+ * > Infura API keys are available for free at: infura.io/register
+ *
+ * $ npm install truffle-hdwallet-provider@V5
+ *
+ */
+
+// const HDWallet = require('truffle-hdwallet-provider');
+// const mnemonic = "twelve-secret-words-only-you-know-for-your-wallet";
+// const infuraKey = "fj4jll3k.....";
+
 module.exports = {
-  // See <http://truffleframework.com/docs/advanced/configuration>
-  // to customize your Truffle configuration!
+  /**
+   * Networks define how you connect to your ethereum client and let you set the
+   * defaults web3 uses to send transactions. If you don't specify one truffle
+   * will spin up a development blockchain for you on port 9545 when you
+   * run `develop` or `test`. You can tell a truffle command to use a specific
+   * network from the command line, e.g
+   *
+   * $ truffle test --network <network-name>
+   *
+   * Everything is optional and any fields you leave out will be filled in with
+   * truffle's default settings.
+   */
+
+  /*
+  networks: {
+    // Useful for testing and general development
+    development: {
+      host: "127.0.0.1",  // Localhost (default)
+      port: 8545,         // Standard Ethereum port (default for defined networks)
+      network_id: "*",    // Any network (default)
+    },
+
+    // Useful if you launch your client locally at the command line yourself.
+    // Examples: ganache-cli, geth, parity, or ...
+    advanced: {
+      port: 8777,            // Custom port
+      network_id: "1342",    // Custom network
+      gas: 8500000,          // Gas sent with each transaction (default: ~6700000)
+      gasPrice: 20000000000  // 20 gwei (in wei) (default: 100 gwei)
```

Comment:
Can we configure the `"estimate"` here, or is that only in truffle-contract?

Reply:
Not yet. Would be nice though. I guess it would be gasMultiplier (coefficient) and even autoGas (boolean).